dotnet/src/dotnetcore/Providers/Messaging/GXAzureServiceBus/ServiceBusMessageBrokerProvider.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections;
+using GeneXus.Messaging.Common;
+using GeneXus.Utils;
+
+namespace GeneXus.Messaging.GXAzureServiceBus
+{
+	public class ServiceBusMessageBrokerProvider
+	{
+		public MessageQueue Connect(string queueName, string connectionString, out GXBaseCollection<SdtMessages_Message> errorMessages, out bool success)
+		{
+			MessageBrokerProvider messageBrokerProvider = new MessageBrokerProvider();
+			GXProperties properties = new GXProperties();
+			properties.Add(PropertyConstants.MESSAGEBROKER_AZURESB_QUEUENAME, queueName);
+			properties.Add(PropertyConstants.MESSAGEBROKER_AZURESB_CONNECTIONSTRING, connectionString);
+
+			MessageQueue messageQueue = messageBrokerProvider.Connect(PropertyConstants.AZURESERVICEBUS, properties, out GXBaseCollection<SdtMessages_Message> errorMessagesConnect, out bool successConnect);
+			errorMessages = errorMessagesConnect;
+			success = successConnect;
+			return messageQueue;
+		}
+
+		public MessageQueue Connect(string topicName, string subcriptionName, string connectionString, out GXBaseCollection<SdtMessages_Message> errorMessages, out bool success)
+		{
+			MessageBrokerProvider messageBrokerProvider = new MessageBrokerProvider();
+			GXProperties properties = new GXProperties();
+			properties.Add(PropertyConstants.MESSAGEBROKER_AZURESB_QUEUENAME, topicName);
+			properties.Add(PropertyConstants.MESSAGEBROKER_AZURESB_SUBSCRIPTION_NAME, subcriptionName);
+			properties.Add(PropertyConstants.MESSAGEBROKER_AZURESB_CONNECTIONSTRING, connectionString);
+
+			MessageQueue messageQueue = messageBrokerProvider.Connect(PropertyConstants.AZURESERVICEBUS, properties, out GXBaseCollection<SdtMessages_Message> errorMessagesConnect, out bool successConnect);
+			errorMessages = errorMessagesConnect;
+			success = successConnect;
+			return messageQueue;
+		}
+		public MessageQueue Connect(string queueName, string connectionString, bool sessionEnabled, GxUserType receiverOptions, string senderIdentifier, out GXBaseCollection<SdtMessages_Message> errorMessages, out bool success)
+		{
+
+			MessageBrokerProvider messageBrokerProvider = new MessageBrokerProvider();
+			ReceiverOptions options = TransformGXUserTypeToReceiverOptions(receiverOptions);
+
+			GXProperties properties = new GXProperties();
+			properties.Add(PropertyConstants.MESSAGEBROKER_AZURESB_QUEUENAME, queueName);
+			properties.Add(PropertyConstants.MESSAGEBROKER_AZURESB_CONNECTIONSTRING, connectionString);
+			properties.Add(PropertyConstants.SESSION_ENABLED, sessionEnabled.ToString());
```

Comment:
It seems that the Connect code is the same code in the 4 Connect methods. Maybe we can do a minor refactoring?

Reply:
It's not exactly the same in fact.. it has some differences that's why I couldn't use a common method..